tests/models/test_base.py

```diff
@@ -0,0 +1,367 @@
+                      f'{diskfile.local_path}/{diskfile.filepath}_2.dat' ]
+    with open( paths[0], 'rb' ) as ifp:
+        assert md5sum1 == hashlib.md5( ifp.read() ).hexdigest()
+    with open( paths[1], 'rb' ) as ifp:
+        assert md5sum2 == hashlib.md5( ifp.read() ).hexdigest()
+    with open( f'{archivebase}{diskfile.filepath}_1.dat', 'rb' ) as ifp:
+        assert md5sum1 == hashlib.md5( ifp.read() ).hexdigest()
+    with open( f'{archivebase}{diskfile.filepath}_2.dat', 'rb' ) as ifp:
+        assert md5sum2 == hashlib.md5( ifp.read() ).hexdigest()
+
+    # Make sure we can delete without purging the archive
+    diskfile.remove_data_from_disk( purge_archive=False )
+    assert diskfile.filepath_extensions == [ '_1.dat', '_2.dat' ]
+    assert diskfile.md5sum_extensions == [ uuid.UUID(md5sum1), uuid.UUID(md5sum2) ]
+    assert diskfile.md5sum == None
+    paths = diskfile.get_fullpath()
+    assert paths == [ f'{diskfile.local_path}/{diskfile.filepath}_1.dat',
+                      f'{diskfile.local_path}/{diskfile.filepath}_2.dat' ]
+    with pytest.raises( FileNotFoundError ):
+        ifp = open( paths[0], 'rb' )
+        ifp.close()
+    with pytest.raises( FileNotFoundError ):
+        ifp = open( paths[1], 'rb' )
+        ifp.close()
+    with open( f'{archivebase}{diskfile.filepath}_1.dat', 'rb' ) as ifp:
+        assert md5sum1 == hashlib.md5( ifp.read() ).hexdigest()
+    with open( f'{archivebase}{diskfile.filepath}_2.dat', 'rb' ) as ifp:
+        assert md5sum2 == hashlib.md5( ifp.read() ).hexdigest()
+
+    # Make sure we can get the file back from the archive
+    paths = diskfile.get_fullpath( nofile=False )
+    assert paths == [ f'{diskfile.local_path}/{diskfile.filepath}_1.dat',
+                      f'{diskfile.local_path}/{diskfile.filepath}_2.dat' ]
+    with open( paths[0], 'rb' ) as ifp:
+        assert md5sum1 == hashlib.md5( ifp.read() ).hexdigest()
+    with open( paths[1], 'rb' ) as ifp:
+        assert md5sum2 == hashlib.md5( ifp.read() ).hexdigest()
+
+    # Delete it all
+    diskfile.remove_data_from_disk( purge_archive=True )
+    assert diskfile.filepath_extensions == [ '_1.dat', '_2.dat' ]
+    assert diskfile.md5sum_extensions == [ None, None ]
+    assert diskfile.md5sum == None
+    paths = diskfile.get_fullpath()
+    assert paths == [ f'{diskfile.local_path}/{diskfile.filepath}_1.dat',
+                      f'{diskfile.local_path}/{diskfile.filepath}_2.dat' ]
+    with pytest.raises( FileNotFoundError ):
+        ifp = open( paths[0], 'rb' )
+        ifp.close()
+    with pytest.raises( FileNotFoundError ):
+        ifp = open( paths[1], 'rb' )
+        ifp.close()
+    with pytest.raises( FileNotFoundError ):
+        ifp = open( f'{archivebase}{diskfile.filepath}_1.dat', 'rb' )
+        ifp.close()
+    with pytest.raises( FileNotFoundError ):
+        ifp = open( f'{archivebase}{diskfile.filepath}_2.dat', 'rb' )
+        ifp.close()
+
```

Comment:
How about putting this delete it all into a `finally` clause? You may need to check that diskfile is not None and stuff like that (for cases where there's a failure very early in the test).